File: Sources/LiveViewNative/UploadConfig.swift
//
//  File.swift
//  
//
//  Created by Carson.Katri on 5/30/23.
//

import Foundation

public struct UploadConfig: Codable {
    let name: String
    let accept: String
    let ref: String
    let maxEntries: Int
    let chunkSize: Int
}

File: lib/live_view_native_swift_ui/types/upload_config.ex
defmodule LiveViewNativeSwiftUi.Types.UploadConfig do
  @derive Jason.Encoder
  defstruct [
    :name,
    :accept,
    :ref,
    :max_entries,
    :chunk_size
  ]

  use LiveViewNativePlatform.Modifier.Type
  def type, do: :map

  def cast(upload_config) when is_struct(upload_config, Phoenix.LiveView.UploadConfig),
    do: {:ok, struct(__MODULE__, Map.from_struct(upload_config))}
  def cast(_), do: :error
end
